### PR TITLE
Zarr hamt decryption

### PR DESCRIPTION
**Pull request by eloramirez1356**
This PR adds the logic to decrypt hamt zarr files encrypted using gridded-etl-tools using the marketplace, concretely [this dclimate-monorepo PR](https://github.com/dClimate/dclimate-monorepo/pull/1050)

I merged [Helia adaptation to zarr files using hamt](https://github.com/dClimate/zarr-ipfs/pull/4) changes and then apply the decryption logic, so this previous PR should be merged before this one.

Important notes:
- I use sodium library for the decryption and I obtain it directly from the frontend, because I tried to install the package in this repo but I could not build it properly. Nevertheless, it is working, so in case we dont like this approach we can research more.

### CODE REVIEW

**Comment by TheGreatAlgo** [comment]
probably better to leave sodium out so others can use their own decryption took they prefer later. this works good. Seems like it has all the changes from [the previous PR](https://github.com/dClimate/zarr-ipfs/pull/4)? if so we could just merge this one in and close the other 3 PRs?

**Review by TheGreatAlgo — APPROVED** [review]

**Review by TheGreatAlgo — APPROVED** [review]
nice removing of any in the code

**Comment by eloramirez1356** [comment]
> probably better to leave sodium out so others can use their own decryption took they prefer later. this works good. Seems like it has all the changes from https://github.com/dClimate/zarr-ipfs/pull/4? if so we could just merge this one in and close the other 3 PRs?

I adapted it for sodium in this case because I call one of the arguments sodium library and also we use specific methods of sodium library in the code too (frontend). So I preferred that and its working, but we could try to improve it for making it more dynamic to achieve what you mention of use different kind of libraries. But for now works.

Related to the PRs, yes. As we are using commits to point different parts of the code (non-hamt and hamt), it would be possible to use the commit id to point to the code we want in the branch.

#3  This one is used for non-hamts. 
#4  and this PR used for hamts.

Sorry as I did not received any notification from this repo.

File: dist/types/storage/ipfsStore.d.ts [hunk 1]
```diff
@@ -1,13 +1,30 @@
 import { AsyncStore } from "./types";
+<<<<<<< HEAD
```
Comment (Faolain) [inline dist/types/storage/ipfsStore.d.ts]: looks like a merge commit made it in here @eloramirez1356

**Pull request merged**